pkgs/development/python-modules/binwalk/default.nix

```
@@ -29,7 +31,7 @@ buildPythonPackage {
     sha256 = "1bxgj569fzwv6jhcbl864nmlsi9x1k1r20aywjxc8b9b1zgqrlvc";
   };
 
-  propagatedBuildInputs = [ zlib xz ncompress gzip bzip2 gnutar p7zip cabextract lzma pycrypto ]
+  propagatedBuildInputs = [ zlib xz ncompress gzip bzip2 gnutar p7zip cabextract cramfsswap cramfsprogs lzma pycrypto ]
   ++ stdenv.lib.optional visualizationSupport pyqtgraph;
```

I have not checked the other tools. I suspect they also should be in PATH rather than
`propagatedBuildInputs`.

```suggestion
  propagatedBuildInputs = [ zlib xz ncompress gzip bzip2 gnutar p7zip cabextract lzma pycrypto ]
  ++ stdenv.lib.optional visualizationSupport pyqtgraph;
  
   makeWrapperArgs = ["--prefix" "PATH" ":" (stdenv.lib.makeBinPath [ cramfsswap cramfsprogs ])];
```

---

I don't understand what this addition is supposed to do:
When I `nix-shell --pure -p 'python3.withPackages(ps: with ps; [binwalk])'`, the `$PATH` is not changed whether I this line (`makeWrapperArgs`...) is present in the expression or not.

---

@jonringer, you made a similar proposition below: What is that supposed to do?

I am not too sure I am testing that correctly :confused:: I don't see any difference when I am adding the aforementioned line:
```bash
# cramfsswap is still not available via the binwalk package
$ nix-shell --pure -p 'python3.withPackages(ps: with ps; [binwalk])' --run "cramfsswap"
/tmp/nix-shell-13495-0/rc: line 1: cramfsswap: command not found

# while it works when imported itself
$ nix-shell --pure -p cramfsswap --run "cramfsswap"
Usage: cramfsswap <in> <out>
```

---

Poke @Mic92 , @jonringer .

---

@Pamplemousse you don't want to change the PATH of your nix-shell but the PATH of the `binwalk` executable. If you execute binwalk from the nix store directly without nix-shell, than `cramfs` won't be in PATH. Note that nix-shell also packs propagated build inputs into your PATH, however those are not proper runtime dependencies but build dependencies instead.

---

Does this makes sense?

---

@Mic92 , thanks for the explanation. So, if I understand correctly:
Using the `makeWrapperArgs ...` allows to solve the error (aforementioned in the first comment) both when invoking `binwalk` from a `nix-shell`, but also from the executable in the nix store.

As for the second part of your remark, I am not quite sure I get the full implications of what you say:
  * So, adding `cramfsswap cramfsprogs` to the `propagatedBuildInputs` do not make them runtime dependencies... Is it only incidental that without `makeWrapperArgs`, and using `nix-shell` (because they are added to its `PATH`) the error was solved when I tested?
  * I kinda hoped those binaries to be available in the shell when importing `binwalk`, is that an over-expectation? (i.e. it is "normal" for import of python modules not to expose in the `nix-shell` the binaries they rely on)